ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V114__activate_string_interning_for_parties_and_templates.sql

```diff
@@ -0,0 +1,975 @@
+     'Verification failed. Original and interned ' || table_name || ' table size mismatch: ' || original_size::TEXT || ' / ' || new_size::TEXT;
+   EXECUTE
+   'WITH uninterned AS (' ||chr(10)||
+   '    SELECT ' ||chr(10)||
+   '      ' || column_select_clause(template_id_column, 'substring(string_interning.external_string from 3)') || ' AS tid,' ||chr(10)||
+   '      ' || uninterned_parties_select_clause(party_array_column_a) || ' AS pa,' ||chr(10)||
+   '      ' || uninterned_parties_select_clause(party_array_column_b) || ' AS pb,' ||chr(10)||
+   '      ' || uninterned_parties_select_clause(party_array_column_c) || ' AS pc,' ||chr(10)||
+   '      ' || uninterned_parties_select_clause(party_array_column_d) || ' AS pd,' ||chr(10)||
+   '      ' || uninterned_parties_select_clause(party_array_column_e) || ' AS pe,' ||chr(10)||
+   '      ' || join_column ||chr(10)||
+   '    FROM ' || table_name || '_interned ' || template_id_join_clause(template_id_column) ||chr(10)||
+   '  ),' ||chr(10)||
+   '  original AS (' ||chr(10)||
+   '    SELECT ' ||chr(10)||
+   '      ' || column_select_clause(template_id_column, 'template_id') || ' AS tid,' ||chr(10)||
+   '      ' || ordered_parties_select_clause(party_array_column_a) || ' AS pa,' ||chr(10)||
+   '      ' || ordered_parties_select_clause(party_array_column_b) || ' AS pb,' ||chr(10)||
+   '      ' || ordered_parties_select_clause(party_array_column_c) || ' AS pc,' ||chr(10)||
+   '      ' || ordered_parties_select_clause(party_array_column_d) || ' AS pd,' ||chr(10)||
+   '      ' || ordered_parties_select_clause(party_array_column_e) || ' AS pe,' ||chr(10)||
+   '      ' || join_column ||chr(10)||
+   '    FROM ' || table_name ||chr(10)||
+   '  )' ||chr(10)||
+   'SELECT array_agg(ten_mismatch.' || join_column || '::TEXT)' ||chr(10)||
+   'FROM (' ||chr(10)||
+   '  SELECT ' || join_column || ' FROM original AS o INNER JOIN' ||chr(10)||
+   '         uninterned AS n USING (' || join_column|| ')' ||chr(10)||
+   '  WHERE o.* != n.*' ||chr(10)||
+   '  LIMIT 10' ||chr(10)||
+   ') ten_mismatch'
+   INTO mismatch_event_seq_ids;
+   ASSERT
+       mismatch_event_seq_ids IS NULL,
+       'Verification failed for ' || table_name ||'. offending id-s (maximum 10): [' || array_to_string(mismatch_event_seq_ids, ', ') || ']';
+   RETURN 'VERIFIED';
+END
+$$;
+
+-- disable hash and merge join to bias the planner to use lookups against the interning table
+SET enable_hashjoin = off;
+SET enable_mergejoin = off;
+
+
+
+
+
+SELECT 'Interning data migration - Migrating party_entries...';
+
+-- add party_id to party_entries
+ALTER TABLE party_entries
+  ADD COLUMN party_id INTEGER;
+
+-- populate internal party id-s
+UPDATE party_entries
+  SET party_id = string_interning.internal_id
+FROM string_interning
+WHERE string_interning.external_string = 'p|' || party_entries.party;
+
+-- add index on party_id
+CREATE INDEX idx_party_entries_party_id_and_ledger_offset ON party_entries(party_id, ledger_offset);
```

Comment:
Missing verification.

What do you think about adding the `NOT NULL` constraint on `party_id` after the `UPDATE`? That would serve as verification and important documentation.

Reply:
We can't unfortunately. This is the same table tracking failed party-additions, where it might be null. (I wanted though ;) this is how I figured)

Reply:
Ah good point. Resolving then.